From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jon Hunter <jonathanh@nvidia.com>
Date: Tue, 9 Jun 2026 09:26:13 +0000
Subject: [PATCH 249/254] Revert "NVIDIA: SAUCE: firmware: tegra: bpmp: add die
 specific debugfs nodes"

This reverts commit 439459f722425778c0c9a2b07fc20b95ea43fa99 so that we
can apply the latest changes that have been prepared for upstream.

Bug 4466065
Bug 6095936

Change-Id: I6a4b41955306c8e6c6aff7254276801de55cc91c
Signed-off-by: Jon Hunter <jonathanh@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3640213
Reviewed-by: svc-mobile-coverity <svc-mobile-coverity@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
APS-Inflight: svcenomis <svcenomis@nvidia.com>
(cherry picked from commit 7d852435b0f6f69e0e2a1e2505f86a8aadf7f9db)

diff --git a/drivers/firmware/tegra/bpmp-debugfs.c b/drivers/firmware/tegra/bpmp-debugfs.c
index 5362e3ce51929..4221fed70ad48 100644
--- a/drivers/firmware/tegra/bpmp-debugfs.c
+++ b/drivers/firmware/tegra/bpmp-debugfs.c
@@ -772,8 +772,6 @@ static int bpmp_populate_debugfs_shmem(struct tegra_bpmp *bpmp)
 int tegra_bpmp_init_debugfs(struct tegra_bpmp *bpmp)
 {
 	struct dentry *root;
-	struct dentry *symlink = NULL;
-	char tmp[10];
 	bool inband;
 	int err;
 
@@ -782,19 +780,10 @@ int tegra_bpmp_init_debugfs(struct tegra_bpmp *bpmp)
 	if (!inband && !tegra_bpmp_mrq_is_supported(bpmp, MRQ_DEBUGFS))
 		return 0;
 
-	if (dev_to_node(bpmp->dev) == NUMA_NO_NODE) {
-		root = debugfs_create_dir("bpmp", NULL);
-	} else {
-		snprintf(tmp, sizeof(tmp), "bpmp%d", dev_to_node(bpmp->dev));
-		root = debugfs_create_dir(tmp, NULL);
-	}
-
+	root = debugfs_create_dir("bpmp", NULL);
 	if (IS_ERR(root))
 		return -ENOMEM;
 
-	if (dev_to_node(bpmp->dev) == 0)
-		symlink = debugfs_create_symlink("bpmp", NULL, "bpmp0");
-
 	bpmp->debugfs_mirror = debugfs_create_dir("debug", root);
 	if (IS_ERR(bpmp->debugfs_mirror)) {
 		err = -ENOMEM;
@@ -808,10 +797,8 @@ int tegra_bpmp_init_debugfs(struct tegra_bpmp *bpmp)
 		err = bpmp_populate_debugfs_shmem(bpmp);
 
 out:
-	if (err < 0) {
-		debugfs_remove(symlink);
+	if (err < 0)
 		debugfs_remove_recursive(root);
-	}
 
 	return err;
 }
-- 
2.43.0

